src/solver/newton/newton.hpp

```diff
@@ -34,8 +34,33 @@ namespace newton {
  * @{
  */
 
-static constexpr int MAX_ITER = 1e3;
-static constexpr double EPS = 1e-12;
+static constexpr int MAX_ITER = 50;
```

Comment:
Don't have relevant expertise and hence curious about which aspects allow much lower max iterations.

Reply:
NEURON uses 20.

If Newton converges, the number of `0`s in the error doubles with each iteration, i.e. from 0.1 error it takes `3` iterations to hit `1e-14`. Unlike stochastic gradient descent, if it doesn't converge, there's no nothing to make it eventually converge.